addons/repair/functions/fnc_doRepair.sqf

```diff
@@ -31,9 +31,23 @@ _hitPointDamage = _hitPointDamage max ([_unit] call FUNC(getPostRepairDamage));
 // raise event to set the new hitpoint damage
 ["setVehicleHitPointDamage", _vehicle, [_vehicle, _hitPoint, _hitPointDamage]] call EFUNC(common,targetEvent);
 
+// Repair the rest in the group (no specific damage, main hitpoint is enough)
+_hitpointGroup = configFile >> "CfgVehicles" >> typeOf _vehicle >> QGVAR(hitpointGroup);
+_hitpointGroup = if (isArray _hitpointGroup) then {getArray _hitpointGroup} else {[]};
+if (count _hitpointGroup > 0) then {
```

Comment:
What do you mean? The lead hitpoint (first in the config array) will get repaired only as calculated, while everything else in it fully as it doesn't matter if they get repaired fully or not really, at least not the the user directly (you wouldn't want to group glass and engine together for example).